mongosql/src/schema/definitions.rs

```diff
@@ -1113,29 +1113,40 @@ impl Schema {
         use Schema::*;
 
         match (&self, &other) {
-            // We currently disallow arrays and documents in comparisons.
-            (Array(_), _) | (_, Array(_)) => Not,
-            (Document(_), _) | (_, Document(_)) => Not,
-
             // Comparing with an Any schema will always result in a May.
             (Any, _) | (_, Any) => May,
 
+            // Any type is comparable to null
+            (Atomic(crate::schema::Atomic::Null), _) | (_, Atomic(crate::schema::Atomic::Null)) => {
+                Must
+            }
```

Comment:
We note this everywhere and intend for this to be true, but relied on it being implicit before. I made it explicit since now it applies to everything else (Arrays and Documents included).